Move token sending to authorization header instead of body parameter

filmaj opened the pull request:
Fixes #1132

I also removed some trailing whitespace because that's the kind of nitpicky dude I am

filmaj commented on packages/web-api/src/WebClient.spec.js:
```diff
@@ -521,9 +522,6 @@ describe('WebClient', function () {
             assert.include(file, { fieldname: 'someBinaryField', filename: 'train.jpg' });
-
-            assert.lengthOf(parts.fields, 1);
-            assert.deepInclude(parts.fields, { fieldname: 'token', value: token });
```
With the token no longer a part of the request body, this part needed removal.

seratch commented on packages/web-api/src/WebClient.ts:
```diff
@@ -183,6 +183,9 @@ export class WebClient extends Methods {
     }
 
+    // eslint-disable-next-line no-param-reassign
+    if (!headers.Authorization) headers.Authorization = `Bearer ${this.token}`;
```
`this.token` can be absent. For instance, some API calls (such as oauth.v2.access with code parameter, openid.connect.token with code parameter, api.test) do not require token in the authorization header. Can you update the logic to check if `this.token` exists too?

seratch commented on packages/web-api/src/WebClient.spec.js:
```diff
@@ -425,7 +426,7 @@ describe('WebClient', function () {
-        .post(/api/, 'token=xoxb-faketoken&team_id=T12345678&foo=stringval&bar=42&baz=false')
+        .post(/api/, 'team_id=T12345678&foo=stringval&bar=42&baz=false')
         .reply(200, { ok: true });
       return this.client.apiCall('method', { foo: 'stringval', bar: 42, baz: false, team_id: 'T12345678' })
```
Can we override the token in the authorization header if a developer passes a token like `this.client.apiCall('method', { token: 'xoxb-' })`? Switching tokens is not a rare use case. For instance, the singleton `WebClient` does not hold any token and your app utilizes multiple user tokens for user related API calls.

seratch reviewed (changes requested):
I like the change! Can you check the two points I mentioned in comments?

filmaj commented:
@seratch I believe I have addressed your comments, and also added another test for the Authorization header overriding behaviour via `apiCall()` options parameter.

seratch reviewed (approved):
@filmaj LGTM! Having unit tests this way makes reviews easier 👍

stevengill reviewed (approved):
Looks great!
